src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -120,13 +121,19 @@ def resolve(self, root_reqs, check_supported_wheels):
             self._result = resolver.resolve(
                 requirements, max_rounds=try_to_avoid_resolution_too_deep,
             )
-
         except ResolutionImpossible as e:
             error = self.factory.get_installation_error(e)
             six.raise_from(error, e)
 
+        return self._make_req_set(
+            self._result.mapping.values(),
+            check_supported_wheels,
+        )
+
+    def _make_req_set(self, candidates, check_supported_wheels):
```

Comment:
Oh hey, this doesn't even need `self`! I suggest making this an `@staticmethod`.

Reply:
This uses `self.factory` a while later on (unlike GitHub, my editor highlights `self`, which I've always considered to be an annoying feature until today):

https://github.com/pypa/pip/blob/191f96229a8638be1fb71433e474aa25c9d1feeb/src/pip/_internal/resolution/resolvelib/resolver.py#L149-L152